File: fronted/app/src/main/java/com/asflum/cashewregister/google/gmail/GmailAccessManager.kt
package com.asflum.cashewregister.google.gmail

import android.content.Context
import com.asflum.cashewregister.google.auth.GmailBackendAuth
import com.asflum.cashewregister.google.auth.GoogleAuthHandler

object GmailAccessManager {

    suspend fun authenticateAndSetup(context: Context): Result<String> {
        val tokenResult = GoogleAuthHandler.getUserIdToken(context)
        if (tokenResult.isFailure) return tokenResult

        val idToken = tokenResult.getOrNull() ?: return Result.failure(Exception("No se pudo obtener el token del usuario."))

        val backendResult = GmailBackendAuth.setupGmailAccess(context, idToken)
        if (backendResult.isFailure) return backendResult

        return Result.success(idToken)
    }
}

File: frontend/lib/services/gmail_access_manager.dart
import 'package:expense_tracker/utils/result.dart';
import 'package:expense_tracker/services/google_auth_handler.dart';
import 'package:expense_tracker/services/gmail_backend.dart';

class GmailAccessManager {
  static Future<Result<String>> authenticateAndSetup() async {
    // Simulación del flujo de autenticación
    final tokenResult = await GoogleAuthHandler.getUserIdToken();

    if (tokenResult.isFailure) {
      return tokenResult;
    }

    final idToken = tokenResult.getOrNull()!;
    
    final backendResult = await GmailBackend().setupGmailAccess(idToken);
    if (backendResult.isFailure) {
      return backendResult;
    }

    return Result.success(idToken);
  }
}
